### PR TITLE
Improve particle property interpolation

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Use Bezier interpolation instead of cubic so that values don't overshoot.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c56ac246024162970e3cd63db2dc7f02eb0bb57b available at: https://gist.github.com/90a82627ef02889ef108<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5954.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5954.dmg

File: libraries/shared/src/Interpolate.cpp [hunk 1]
```diff
@@ -11,41 +11,46 @@
...
-
-    return (a0 * uuu + a1 * uu + a2 * u + a3);
+float Interpolate::bezierInterpolate(float y1, float y2, float y3, float u) {
+    // https://en.wikipedia.org/wiki/Bezier_curve
+    return (1.0f - u) * (1.0f - u) * y1 + 2.0f * (1.0f - u) * u * y2 + u * u * y3;
 }
 
-float Interpolate::cubicInterpolate3Points(float y1, float y2, float y3, float u) {
-    float y0, y4;
+float Interpolate::interpolate3Points(float y1, float y2, float y3, float u) {
```
Comment (AndrewMeadows) [inline libraries/shared/src/Interpolate.cpp]: Should we assert here that **u** is inside the range [0.0, 1.0]?

**Comment by AndrewMeadows** [comment]
What is the best way to test this?

**Comment by ctrlaltdavid** [comment]
For testing, there's a modifed version of the particlesTest.js script at the following URL: http://ctrlaltstudio.com/downloads/hifi/scripts/particlesTest.js
Click on the box that's rendered to cycle the emitter through a selection of particle property settings.
If you run it in an official build you'll see the particle radius unexpecdedly ringing in some cases (esp. the "Bad cubic example" test).
If you run this PR build you'll see the particle radius interpolating smoothly as one would expect.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e56d2d9b434622a63e1485ea5e16b945202d2ba6 available at: https://gist.github.com/90a82627ef02889ef108<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5954.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5954.dmg

**Pull request merged**